### PR TITLE
Add setUserData() functions to allow sending own data to templates

### PR DESCRIPTION
**Pull request by Laurent3170**

### CODE REVIEW

**Comment by Laurent3170** [comment]
Or maybe better, set only one function `setUserData()` per Controller and let the user return a multi-array accessible in Twig via `data.show.gravatar` or whatever he/she needs. I think this solution is simplier.
I don't know if I can (and how) update the PR of if I must close it and create a new one.

File: Controller/ChangePasswordController.php [hunk 1]
```diff
@@ -86,6 +86,15 @@ public function changePasswordAction(Request $request)
 
         return $this->render('@FOSUser/ChangePassword/change_password.html.twig', array(
             'form' => $form->createView(),
+            'data' => $this->setUserData(),
         ));
     }
+
+    /*
+     * Function to override to send data to the template
```
Comment (core23) [inline Controller/ChangePasswordController.php]: This is a method

**Comment by core23** [comment]
What's the actual purpose of this PR? If you need custom data, you can still extend the controllers

**Comment by Laurent3170** [comment]
Yes You're right there is no explanation, it's in #2146

File: Controller/ChangePasswordController.php (hunk 1 above)
Comment (Laurent3170) [inline Controller/ChangePasswordController.php]: In french we call it 'Fonction', I have translated it to 'function' but you're right in english it's a 'Method'

Comment (core23) [inline Controller/ChangePasswordController.php]: In German, it's actual called `Funktion` :D

A method is a kind of function used in the context of a class/object.

Comment (Laurent3170) [inline Controller/ChangePasswordController.php]: Yes, I don't have the whole php wording yet ;-) I have learned Symfony (+ Git, Github, etc.) by myself (with tutorials) but it's better to use the good words. Thanks

**Comment by Laurent3170** [comment]
I have updated with one method in my fork https://github.com/Laurent3170/FOSUserBundle/commit/8e4eefa191a113fdef3b6cd6c4b2a71125df27a2 but apart re-submit another PR I don't know how to update this PR.
It's a bit messy... but that's the way to learn...
So I prefer stop there waiting further discussion :-)

**Comment by Laurent3170** [comment]
Hi, any news about that PR, cause I "need" it. Of course, it's not because I need it that it has to be merged ;-).
I have tried to implement the "repositories" inside composer with the following, but it doesn't work...
```
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/975L/FOSUserBundle"
        }
    ],
    "require": {
        "friendsofsymfony/user-bundle": "dev-userdata"
    }
```
So, apart form re-writing each controller, or maintain a fork by my side, I don't know how to send, easily, data to the Twig templates.

**Comment by Laurent3170** [comment]
Replaced by #2589 which is quite cleaner and better explained :-)

**Pull request closed without merging**